attest/api/src/convert/collateral.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! Convert to/from attest::Collateral
+
+use crate::{attest, ConversionError};
+use mc_attest_verifier_types::prost;
+use mc_sgx_dcap_types::Collateral;
+use mc_util_serial::Message;
+use protobuf::{CodedOutputStream, Message as ProtoMessage};
+
+impl TryFrom<&Collateral> for attest::Collateral {
+    type Error = ConversionError;
+    fn try_from(src: &Collateral) -> Result<Self, Self::Error> {
+        let prost = prost::Collateral::try_from(src)?;
+        let bytes = prost.encode_to_vec();
+        let mut proto = Self::default();
+        proto
+            .merge_from_bytes(&bytes)
+            .expect("failure to merge means prost and protobuf are out of sync");
```

Comment:
This is a try_from so I'm fine if we think an error is better here, It just feels like if this fails then the prost and gRPC protobuf implementations are broken.